doozer/doozerlib/cli/config_tag_rpms.py

```diff
@@ -150,9 +151,34 @@ async def run(self):
                         logger.warning("Build %s is already tagged into %s", build["nvr"], entry.target_tag)
                         break
                     logger.info("Build %s is acceptable for %s", build["nvr"], entry.target_tag)
-                    builds_to_tag[entry.target_tag].add(build["nvr"])
+                    builds_to_tag[entry.target_tag][build["nvr"]] = build
                     break
 
+            if not entry.enforce_same_version:
+                continue
+
+            for tag, nvr_dict in builds_to_tag.items():
+                if not nvr_dict:
+                    continue
+
+                package_names = {b['name'] for b in nvr_dict.values()}
+                if package_names != {'kernel', 'kernel-rt'}:
+                    raise ValueError(f"Unexpected packages found with enforce_same_version: {package_names}")
```

Comment:
There can be case when one of them (either kernel or kernel-rt) is missing, so this would change to "Need 1 build of kernel and 1 build of kernel-rt" or something like that